code/API_definitions/qos-booking-and-assignment.yaml

```diff
@@ -0,0 +1,1756 @@
+        * 'SERVICE_NOT_AVAILABLE' - The requested booking could not be fullfilled because of operator could not support the service area/number of devices/qosProfile in the requested time
+        * 'QOS_PROFILE_NOT_SUPPORTED'  - The requested QoS Profile is not supported by the operator
+        * 'BOKING_UNKNOWN_ERROR' - An unknown error happend while booking
+        * 'BOOKING_COMPLETED' - The requested booking is commpleted successfully. 
+        * 'BOOKING_INVALID'  -  The given bookingId is invalid, malformed, or not available in the system
+        * 'BOOKING_CANCELLED'-  The given bookingId was cancelled by the user and is no more valid. 
+        * 'BOOKING_EXPIRED'  -  The given bookingId is expired and is no more valid
+
+      type: string
+      enum:
+        - DURATION_EXPIRED
+        - NETWORK_TERMINATED
+        - DELETE_REQUESTED
+        - VALIDATION_PENDING
+        - SERVICE_NOT_AVAILABLE
+        - QOS_PROFILE_NOT_SUPPORTED
+        - BOOKING_UNKNWON_ERROR
+        - BOOKING_COMPLETED
+        - BOOKING_INVALID
+        - BOOKING_CANCELLED
+        - BOOKING_EXPIRED
+
+    AssignmentStatusInfo:
+      description: |
+       This schema provides a detailed status of an assignment request
+        * `DEVICE_NOT_FOUND` - One or some of the devices are not found to validate and associate to booking. 
+        * 'DEVICE_UNKNOWN_ERROR' - One, or more, or all of devices are not assigned or not released because of unknown error.
+        * 'DEVICE_DISCLOSURE_ERROR' - Some or all of the devices are not disclosed because of privacy, or legal requirements, or for unknown reasons.
+        * 'QUOTA_EXCEEDED' - The number of devices requested to assign to a booking is more than the allocated totoal for the booking. Only remaining devices are assigned
+        * 'VALIDATION_PENDING' - The device assignment request is pending for validation and verification
+        * 'ASSIGNMENT_COMPLETED' - The request assignment of devices is successfully completed
+        * 'BOOKING_INVALID' - The user requests to release a device from booking, but the booking is invalid
+        * 'BOOKING_CANCELLED' - The user requests to release a device from booking, but the booking is cancelled
+        * 'BOOKING_EXPIRED' - The user requests to release a device from booking, but the booking is expiered and no more active
+        * 'BOOKING_TERMINATED' - For unknown reasons booking is terminated by the operator. A booking callback may be issued. 
+
+      type: string
+      enum:
+        - DEVICE_NOT_FOUND
+        - DEVICE_UNKNOWN_ERROR
+        - DEVICE_DISCLOSURE_ERROR
+        - QUOTA_EXCEEDED
+        - VALIDATION_PENDING
+        - ASSIGNMENT_COMPLETED
+        - BOOKING_INVALID
+        - BOOKING_CANCELLED
+        - BOOKING_EXPIRED
+        - BOOKING_TERMINATED
+
+    StatusSource:
+      description: |
+        Very useful information when a callback is called with full message body

```

Comment:
Can you please explain the aim of this property? What is the relation with callbacks and what means a "full message body" for a callback. I think we don't have anything similar to this in any other API dealing with callbacks